### ISSUE TITLE
Dynamics Tabs

### ISSUE THREAD

**Issue by baden32** [ng-bootstrap/ng-bootstrap#1262]
I'm wondering whether Dynamics Tabs feature will be available in the officiel release of ng-bootstrap.

This feature would be useful to close and open tab (with associated content) dynamically.
Thank you in advance for your answer.
Joël

**Comment by pkozlowski-opensource** [comment]
What do you mean by "Dynamics Tabs" exactly? Please provide more context and real-life examples, otherwise this one is not actionable....

**Comment by baden32** [comment]
A good example could be found in ng2-bootstrap: http://valor-software.com/ng2-bootstrap/index-bs4.html#/tabs#tab-directive

Tab definition is based on an array which can be changed.programmatically This way a tab can be added or removed.

**Comment by pkozlowski-opensource** [comment]
You can generate tabs "dynamically" today (in the sense of the example you've provided): http://plnkr.co/edit/JNgHED0EhqCSEJvkzNCI?p=preview. This is and was always supported.

Having said this I think that this form have limited practical usage as you can't have HTML nor directives in your content. Having HTML could expose you to potential security issues and having directives would mean shipping compiler to a browser (so no AoT).

tl;dr; is that what you are asking for is supported today but IMO it has limited practical value. But at least you can rest assured that things are working as you would expect :-)

Going o close this one as I don't think there is much to do here since things already work as expected.

**Comment by KostyaTretyak** [comment]
@pkozlowski-opensource, by the way, why `*ngFor` not correctly works here? http://plnkr.co/edit/evEvONYUoGO5ohiO6tup?p=preview

In `tab.content` are three items, but print only first.
